### PR TITLE
refactor: sun and moon path now use arc (pivotal #171465151)

### PR DESCRIPTION
**Pull request by glassesaaron**

### CODE REVIEW

**Comment by glassesaaron** [comment]
I've been working on this for a few weeks, so some ideas may be a little disjointed. Open to any and all feedback on how to make this better.

File: app/javascript/components/widgets/weather/moon-phase.jsx [hunk 1]
```diff
@@ -2,43 +2,105 @@ import React from 'react';
...
+    case 'firstQuart':
+    case 'thirdQuart':
+      return `M 0 -${radius} A ${diameter * radiusModifier1} ${diameter *
+        radiusModifier1} 0 0,0 0 ${diameter - radius}`;
+    case 'waxGib':
+    case 'wanCres':
+      return `M 0 -${radius} A ${diameter * radiusModifier1} ${diameter *
+        radiusModifier1} 0 0,1 0 ${diameter -
+        radius} M 0 -${radius} A ${diameter * radiusModifier2} ${diameter *
+        radiusModifier1} 0 0,0 0 ${diameter - radius}`;
```
Comment (chewnoill) [inline app/javascript/components/widgets/weather/moon-phase.jsx]: Maybe these would be easier to read if we split the commands?

```js
return [
 `M 0 -${radius}`,
 `A ${diameter * radiusModifier1} ${diameter *
        radiusModifier1} 0 0,1 0 ${diameter -
        radius}`
...
].join('')

```
Might be easier to extract pieces that can be reused that way

Comment (glassesaaron) [reply app/javascript/components/widgets/weather/moon-phase.jsx]: @chewnoill thanks for the review! I've updated the `moon-phase.jsx` file with your requested changes. When you get a chance could you please let me know if there's anything else I should fix?

File: app/javascript/components/widgets/weather/moon-phase.jsx [hunk 2]
```diff
@@ -2,43 +2,142 @@ import React from 'react';
...
+        `A ${diameter * radiusModifier1} ${diameter * radiusModifier1}`,
+        `0 0,1 0 ${diameter - radius}`,
+      ];
+      break;
+    case 'firstQuart':
+    case 'thirdQuart':
+      path = [
+        `M 0 -${radius}`,
+        `A ${diameter * radiusModifier1} ${diameter * radiusModifier1}`,
+        `0 0,1 0 ${diameter - radius}`,
```
[the remark this answers is not in the document]
Comment (glassesaaron) [reply app/javascript/components/widgets/weather/moon-phase.jsx]: No, you are correct: I will fix this.

File: app/javascript/components/widgets/weather/moon-phase.jsx [hunk 3]
```diff
@@ -2,43 +2,142 @@ import React from 'react';
...
+      break;
+    case 'waxCres':
+    case 'wanGib':
+      path = [
+        `M 0 -${radius}`,
+        `A ${diameter * radiusModifier1} ${diameter * radiusModifier1}`,
+        `0 0,0 0 ${diameter - radius}`,
+        `M 0 -${radius}`,
+        `A ${diameter * radiusModifier2} ${diameter * radiusModifier1}`,
+        `0 0,1 0 ${diameter - radius}`,
```
[the remark this answers is not in the document]
Comment (glassesaaron) [reply app/javascript/components/widgets/weather/moon-phase.jsx]: This one actually is different: one uses `radiusModifier1`, and the other uses `radiusModifer2`, so it should remain as-is

**Comment by glassesaaron** [comment]
@cbarber I have updated the code according to your comments. The only thing I didn't address was turning the `keyframes` animation into a styled component with props, as this animation and percent is used in both the sun and child moon components and I felt the refactor did not necessarily help. Please let me know if you want me to make this update as well though and I can change it!

**Review by chewnoill — APPROVED** [review]
🚀

**Comment by cbarber** [comment]
This broke with a styled-components upgrade for a security vulnerability. @Aceba1 is grabbing this to rebase.

**Comment by cbarber** [comment]
@glassesaaron sorry a year late, but your work is merged now :tada: https://github.com/mojotech/helios2/pull/211

**Comment by glassesaaron** [comment]
@cbarber 🙏

**Pull request closed without merging**